website/src/components/Edit/SequencesForm.tsx

```diff
@@ -1,117 +1,195 @@
+import { PLAIN_SEGMENT_KIND, VirtualPlainSegmentFile } from '../Submission/FileUpload/fileProcessing.ts';
 
-function generateAndDownloadFastaFile(
-    accessionVersion: string,
-    sequenceData: string,
-    segmentKey?: string,
-    isSingleSegment: boolean = false,
-) {
-    const fileName =
-        isSingleSegment || !segmentKey ? `${accessionVersion}.fasta` : `${accessionVersion}_${segmentKey}.fasta`;
-
-    const header = isSingleSegment || !segmentKey ? accessionVersion : `${accessionVersion}_${segmentKey}`;
-
-    const fileContent = `>${header}\n${sequenceData}`;
+function generateAndDownloadFastaFile(fastaHeader: string, sequenceData: string) {
+    const trimmedHeader = fastaHeader.replace(/\s+/g, '');
+    const fileContent = `>${trimmedHeader}\n${sequenceData}`;
 
     const blob = new Blob([fileContent], { type: 'text/plain' });
     const url = URL.createObjectURL(blob);
 
     const a = document.createElement('a');
     a.href = url;
-    a.download = fileName;
+    a.download = `${trimmedHeader}.fasta`;
     a.click();
 
     URL.revokeObjectURL(url);
 }
 
+type EditableSequenceFile = {
+    key: string;
+    label: string | null;
+    fastaHeader: string | null;
+    value: string | null;
+    initialValue: string | null;
+};
+
 export class EditableSequences {
-    private constructor(public readonly rows: Row[]) {}
-
-    private static emptyRows(names: string[]): Row[] {
-        return names.map((name) => ({
-            key: name,
-            initialValue: '',
-            value: '',
-            errors: [],
-            warnings: [],
+    private static nextKey = 0;
+
+    private readonly editableSequenceFiles: EditableSequenceFile[];
+    private readonly maxNumberOfRows: number;
+
+    public get rows(): Required<EditableSequenceFile>[] {
+        const rows = this.editableSequenceFiles.map((row, i) => ({
+            ...row,
+            label: row.label ?? `Segment ${i + 1}`,
         }));
+        if (rows.length < this.maxNumberOfRows) {
+            rows.push({
+                label: `Add a segment`,
+                fastaHeader: null,
+                value: null,
+                initialValue: null,
+                key: EditableSequences.getNextKey(),
+            });
+        }
+        return rows;
     }
 
-    private isMultsegmented() {
-        return this.rows.length > 1;
+    private constructor(rows: EditableSequenceFile[], maxNumberOfRows: number) {
+        this.editableSequenceFiles = rows;
+        this.maxNumberOfRows = maxNumberOfRows;
+    }
+
+    isMultiSegmented() {
+        return this.maxNumberOfRows > 1;
+    }
+
+    static invertRecordMulti(obj: Record<string, string | null>): Record<string, string[]> {
+        const inverted: Record<string, string[]> = {};
+
+        for (const key in obj) {
+            const value = obj[key];
+            if (value === null) continue;
+            (inverted[value] ??= []).push(key);
+        }
+
+        return inverted;
```

Comment:
Does that make sense? Can it really happen that one sequence is mapped to several segments?